### PR TITLE
[MM-54242] Improve timestamp accuracy

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR adds a speech detection step prior to transcribing. This was made necessary as `whisper.cpp` doesn't currently support word level timestamps (see https://github.com/ggerganov/whisper.cpp/issues/375) so portions of silence could skew the sync of the final transcription by quite a bit.

The way this works is by first feeding the decoded audio samples to the [Silero VAD model](https://github.com/snakers4/silero-vad) and essentially trim silence from the track, splitting it into separate speech segments before passing it to the Whisper engine for transcription.

@jupenur Apologies for the late addition of a non trivial dependency. Let me know if you have any questions. 

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-54242

### CODE REVIEW

File: cmd/transcriber/call/tracks.go [hunk 1]
```diff
@@ -304,13 +306,77 @@ func (t *Transcriber) transcribeTrack(ctx trackContext) (transcribe.TrackTranscr
...
+			slog.Error("failed to detect speech",
+				slog.String("err", err.Error()),
+				slog.String("trackID", ctx.trackID))
+			continue
+		}
+		slog.Debug("speech detection done", slog.Any("segments", segments))
+
+		for _, seg := range segments {
+			startSampleOff := int(seg.SpeechStartAt * trackOutAudioRate)
+			endSampleOff := int(seg.SpeechEndAt * trackOutAudioRate)
```
Comment (cpoile) [inline cmd/transcriber/call/tracks.go:359]: One request -- can you add some comments for future us about what the units are for `seg.SpeechStartAt` and `startSampleOff`? (That will help in the future when I'm reviewing and remembering what the numbers mean.)

**Review by cpoile — APPROVED** [review]
Awesome (as usual)! :)

File: cmd/transcriber/call/tracks.go (hunk 1 above)
Comment (streamer45) [reply cmd/transcriber/call/tracks.go:359]: Sure but the sample offset is just what it says. It's an index to the first sample where the speech starts (+- some padding), there's no unit really.

File: build/prepare_deps.sh [hunk 2]
```diff
@@ -6,6 +6,8 @@ OPUS_SHA=$2
 WHISPER_VERSION=$3
 WHISPER_SHA=$4
 MODELS=$5
+ONNX_VERSION=$6
+ONNX_SHA=$7
```
Comment (jupenur) [inline build/prepare_deps.sh:10]: You need to actually use this too 😜

**Review by jupenur — CHANGES_REQUESTED** [review]

File: build/prepare_deps.sh (hunk 2 above)
Comment (streamer45) [reply build/prepare_deps.sh:10]: Just testing our security engineer's eye :p
Aha, good catch :)

**Review by jupenur — APPROVED** [review]

**Pull request merged**